config.go

```diff
@@ -9,48 +9,48 @@ import (
 	"unsafe"
 )
 
-// Compilation strategies for wasmtime
+// Strategy is the compilation strategies for wasmtime
 type Strategy C.wasmtime_strategy_t
 
 const (
-	// Wasmtime will automatically pick an appropriate compilation strategy
-	STRATEGY_AUTO Strategy = C.WASMTIME_STRATEGY_AUTO
-	// Force wasmtime to use the Cranelift backend
-	STRATEGY_CRANELIFT Strategy = C.WASMTIME_STRATEGY_CRANELIFT
-	// Force wasmtime to use the lightbeam backend
-	STRATEGY_LIGHTBEAM Strategy = C.WASMTIME_STRATEGY_LIGHTBEAM
+	// StrategyAuto will let wasmtime automatically pick an appropriate compilation strategy
+	StrategyAuto Strategy = C.WASMTIME_STRATEGY_AUTO
+	// StrategyCranelift will Force wasmtime to use the Cranelift backend
+	StrategyCranelift Strategy = C.WASMTIME_STRATEGY_CRANELIFT
+	// StrategyLightbeam will force wasmtime to use the lightbeam backend
+	StrategyLightbeam Strategy = C.WASMTIME_STRATEGY_LIGHTBEAM
 )
 
-// What degree of optimization wasmtime will perform on generated machine code
+// OptLevel decides what degree of optimization wasmtime will perform on generated machine code
 type OptLevel C.wasmtime_opt_level_t
 
 const (
-	// No optimizations will be performed
-	OPT_LEVEL_NONE OptLevel = C.WASMTIME_OPT_LEVEL_NONE
-	// Machine code will be optimized to be as fast as possible
-	OPT_LEVEL_SPEED OptLevel = C.WASMTIME_OPT_LEVEL_SPEED
-	// Machine code will be optimized for speed, but also optimized
+	// OptLevelNone will perform no optimizations
+	OptLevelNone OptLevel = C.WASMTIME_OPT_LEVEL_NONE
+	// OptLevelSpeed will optimize machine code to be as fast as possible
+	OptLevelSpeed OptLevel = C.WASMTIME_OPT_LEVEL_SPEED
+	// OptLevelSpeedAndSize will optimize Machine code for speed, but also optimize
 	// to be small, sometimes at the cost of speed.
-	OPT_LEVEL_SPEED_AND_SIZE OptLevel = C.WASMTIME_OPT_LEVEL_SPEED_AND_SIZE
+	OptLevelSpeedAndSize OptLevel = C.WASMTIME_OPT_LEVEL_SPEED_AND_SIZE
 )
 
-// What sort of profiling to enable, if any.
+// ProfilingStrategy decides what sort of profiling to enable, if any.
 type ProfilingStrategy C.wasmtime_profiling_strategy_t
 
 const (
-	// No profiler will be used
-	PROFILING_STRATEGY_NONE ProfilingStrategy = C.WASMTIME_PROFILING_STRATEGY_NONE
-	// The "jitdump" linux support will be used
-	PROFILING_STRATEGY_JITDUMP ProfilingStrategy = C.WASMTIME_PROFILING_STRATEGY_JITDUMP
+	// ProfilingStrategyNone means No profiler will be used
+	ProfilingStrategyNone ProfilingStrategy = C.WASMTIME_PROFILING_STRATEGY_NONE
+	// ProfilingStrategyJitdump will use the "jitdump" linux support
+	ProfilingStrategyJitdump ProfilingStrategy = C.WASMTIME_PROFILING_STRATEGY_JITDUMP
 )
 
-// Configuration of an `Engine` which is used to globally configure things
+// Config is the configuration of an `Engine` which is used to globally configure things
```

Comment:
This one reads oddly (and so did the former one!). Maybe too much "config"? Can't think of something better though.

Reply:
```
// Config is a configuration for `Engine` which will be used to support other structs globally.
// You can create an favored Engine by NewEngineWithConfig
```
Is it better?

And I think we can rename the `Config` struct to `EngineConfig` because there is a `WasiConfig`

Reply:
I think it should stay `Config` because that's what it's called upstream in `wasmtime`: https://docs.rs/wasmtime/0.17.0/wasmtime/struct.Config.html

If we take a note from the upstream docs, it may look like:
Config holds options used to create an Engine and customize its behavior.

Reply:
I think the upstream's is better.